core/Updates/5.0.0-b1.php

```diff
@@ -110,4 +117,39 @@ private function requiresUpdatedLogVisitTableIndex()
 
         return true;
     }
+
+    /**
+     * Add new 'pages before column' to log conversions
+     *
+     * @return DbAlias\AddColumns
+     */
+    private function getPagesBeforeAddColumn()
+    {
+        return $this->migration->db->addColumns('log_conversion', ['pageviews_before' => "SMALLINT UNSIGNED DEFAULT NULL"]);
+    }
+
+    private function populatePagesBefore(): void
+    {
+        // Abort if host is *.matomo.cloud or *.innocraft.cloud
+        $instanceId = SettingsPiwik::getPiwikInstanceId();
+        if (strpos($instanceId, '.matomo.cloud') !== false || strpos($instanceId, '.innocraft.cloud') !== false) {
```

Comment:
FYI it would actually work for custom domains as we're checking the instanceId and not the matomo url.

There is no process yet for this AFAIK. There could be an issue creation but this may result in a bit of manual work. Or maybe if it's only used in update script, maybe it wouldn't be too bad to have a `isCloud` method and then not execute it.

Reply:
So is it okay to leave this check as is since it's just a update script?  I could adjust it to check if the cloud plugin is loaded instead if that would be more reliable?

Reply:
But don't we need to run the update on cloud as well? Otherwise the calculated numbers for today might not be correct afterwards... Or is the purpose to be able to run the update manually and not all at once?

Reply:
We may handle that manually @sgiehl or ideally, if there are say less than 500 conversions in the last 24 hours or simply generally we could always reprocess the last 500 conversions. This would solve it for many. We should check the performance of the query though ideally as @bx80 suggested

Reply:
@sgiehl I did some testing of the query speed and it's quite fast, 10,000 conversions updated in under 10 seconds and 1,000,000 in under a minute. See my comment [here](https://github.com/matomo-org/matomo/pull/20526#discussion_r1175921101). Since we're only processing the last 48hrs then I think a 10,000 limit should be fast enough for even large on premise users? 

With cloud we'd be running the update on thousands of customers at the same time using the same shared resources, so even if any individual database takes ~30 seconds that could be ~80 hours per 10,000

Would it make sense that if we detect more than 10,000 conversions in the last 48hrs during the upgrade and do not perform the update query then we set an option that can then be used to show a diagnostic warning that shows  the command to run?

Reply:
After discussion in the sprint review it was decided to remove the check since the update query is fast and executed separately per goal/site combination. Cloud will not run the update query at all.